src/v1/wasm_ctx.rs

```diff
@@ -90,6 +90,11 @@ impl WASMArgs {
   pub fn shard_size(&self) -> Option<usize> {
     self.trace_slice_vals.map(|val| val.shard_size())
   }
+
+  /// Get clone of bytecode.
+  pub fn bytecode(&self) -> Vec<u8> {
+    self.program.clone()
```

Comment:
I assume you can't avoid this `clone()` but any specific reason to not return a `&[u8]`?

Reply:
Due to internal API limitations it needs to owned. Once zkEngine's API's have stabilized I can rework the APIs in novanet and we may be able to avoid this clone

Reply:
I'm not overly concerned about avoiding the clone itself; I just thought it was best practice to make sure that users of the API are aware when a `clone()` is happening (i.e., making it explicit in the code).